src/irmin/conf.ml

```diff
@@ -120,3 +120,11 @@ let root =
     ~doc:"The location of the Git repository root."
     ~docs:"COMMON OPTIONS"
     "root" (some string) None
+
+let merge a b =
+  let r = match get a root with
+    | Some _ as root -> root
+    | None -> get b root
```

Comment:
just curious: why do you need to special case `root` here?

Reply:
e.g. is the way how `root` is handled the only difference between `merge` and `union`?

Reply:
This is needed because an empty root configuration can overwrite a valid one when using `union`.

For example, if you call `union a b` where `root` is `Some "path"` for `a` and `None` for `b`, the resulting config will end up with `root` being `None`. This is due to the fact that `root` is an option type and is present in the configuration by default.

To clarify a bit more, when you call `Conf.find config Conf.root` you will get `Some None` if it's not yet set, however other keys will typically return `None`. 

I'm happy to develop this further if you can think of a more generic solution.

Reply:
Ha I see, that makes sense; but that means that any other optional key will be overwritten which is not what we want (I think).

Maybe we should filter the config maps to remove all the `None` in there.  I haven't really though about it, but I think that's the right approach (as there is no way to force optional values to be `None` using the CLI or the config files).

Reply:
What is the best way to filter the config maps? I know there is a filter function for `Map`, but I'm not sure how to check that the value is not `None` since it is represented as `Univ.t`inside the map.

Sorry if this is obvious.

Reply:
That makes sense, thanks!